isthmus/src/test/java/io/substrait/isthmus/utils/UserTypeFactory.java

```diff
@@ -0,0 +1,67 @@
+package io.substrait.isthmus.utils;
+
+import io.substrait.type.Type;
+import io.substrait.type.TypeCreator;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeImpl;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+/**
+ * Utility factory for creating user-defined types implementing {@link RelDataTypeImpl}, correctly.
+ * Used for creating types as part of testing.
+ */
+public class UserTypeFactory {
```

Comment:
There's a lot of boilerplate to implementing a `RelDataType` in Calcite, so I made a factory for them (for testing only).